foundations/src/telemetry/memory_profiler.rs

```diff
@@ -1,24 +1,25 @@
 use super::settings::MemoryProfilerSettings;
 use crate::utils::feature_use;
 use crate::{BootstrapError, BootstrapResult, Result};
+use anyhow::anyhow;
 use anyhow::bail;
-use once_cell::sync::{Lazy, OnceCell};
+use once_cell::sync::OnceCell;
 use std::ffi::{CStr, CString};
 use std::fs::File;
 use std::io::Read;
 use std::os::raw::c_char;
-use std::thread;
+use std::sync::mpsc::{self};
+use std::sync::{Arc, Mutex};
+use std::thread::{spawn, JoinHandle};
 use tempfile::NamedTempFile;
-use tokio::sync::Mutex as AsyncMutex;
-use tokio::task::spawn_blocking;
+use tokio::sync::oneshot;
```

Comment:
Do we really require tokio's `oneshot` or could we use `std` one?

Reply:
Not sure exactly what you're referring to - there isn't a `oneshot` in `std`. There's the [oneshot crate](https://crates.io/crates/oneshot), otherwise, as far as I know, `std` just has its general `mpsc`.

Regardless, I think tokio's `oneshot` is the perfect construct for what we want. Its semantics are preferable over an `mpsc` since we always expect one request to produce exactly one response, which its interface captures perfectly. And I think waiting on an async `Receiver` is preferable over a blocking/synchronous one since the actual work the heap profiler is doing is truly on a separate `std::thread`, so the thread asynchronously handling the HTTP request can theoretically do other things while waiting.

Reply:
> there isn't a oneshot in std.

Oh, I thought there would be, I honestly didn't bother checking. In that case, nevermind.